config-ui/src/components/blueprints/GitlabProjectsSelector.jsx

```diff
@@ -0,0 +1,161 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+import React, { useEffect, useState } from 'react'
+import { Checkbox, Intent, MenuItem, } from '@blueprintjs/core'
+import { MultiSelect } from '@blueprintjs/select'
+
+const GitlabProjectsSelector = (props) => {
+  const {
+    fetchGitlabProjects = () => [],
```

Comment:
This should be refactored to something more generic to the component, `onFetch` or `onProjectsFetch`

```suggestion
    onFetch = () => [],
```

Reply:
fixed

Reply:
I don't the argument props definitions shouldn't be aliased, 

```
    fetchGitlabProjects: onFetch = () => [],
     isFetchingGitlab: isFetching = false,
```

should just be:

```
    onFetch = () => [],
    isFetching: isFetching = false,
```


the alias can be done when GitlabProjectsSelector is implemented (isFetching and fetchGitlabProjects are retrieved from the Gitlab Hook when hook is initialized, and aliased during de-structuring)

```
<GitlabProjectsSelector isFetching={isFetchingGitlab) ...
```

Reply:
ok